File: java/src/main/java/com/sweetmanor/algorithm/graph/Edge.java
package com.sweetmanor.algorithm.graph;

public class Edge {

    public final int u; //起点
    public final int v; //终点

    public Edge(int u, int v) {
        this.u = u;
        this.v = v;
    }

    public Edge reversed() {
        return new Edge(v, u);
    }

    @Override
    public String toString() {
        return u + " -> " + v;
    }

}

File: python/src/com/sweetmanor/algorithm/graph/edge.py
# 《算法精粹》

from dataclasses import dataclass


@dataclass
class Edge:
    u: int  # 起点
    v: int  # 终点

    def reversed(self) -> 'Edge':
        return Edge(self.v, self.u)

    def __str__(self):
        return f'{self.u} -> {self.v}'
